_posts/2018-01-01-futures-0.3.0-alpha.3.md

```diff
@@ -0,0 +1,124 @@
+
+A compatibility layer between 0.3 an 0.1 was developped. It is now possible to convert an 0.3 future into an 0.1 future and vice versa. Similar conversions for streams and sinks are also supported. Additionally it is now possible to run 0.3 futures and async functions on Tokio's executor. We have a dedicated blog post coming up that explains this in more detail.
+
+## Spawning
+
+### `Spawn` trait
+
+The `std::task::Executor` trait is now called `std::task::Spawn`. The purpose of the trait remains unchanged - it is to provide a mechanism for spawning tasks. We noticed that in many implementations the responsibilty of spawning tasks is handled by a type that merely sends the task to an associated executor, not by the task executor type itself. Furthermore, it became clear that the former trait name "`Executor`" didn't properly capture the essence of what it means to be an executor. Fundamentally, a task executor is a type that manages the execution of tasks. There exists no fixed set of methods that a trait could define. Instead, each task executor type offers different methods that make sense for its design goals. This is why we chose to rename the trait to `Spawn` to reflect that it merely defines a method for spawning tasks and makes no specifications about how to run them.
+
+As a consequence of this renaming, some functions on `Context` were adjusted:
+```rust
+impl<'a> Context<'a> {
+    pub fn new(local_waker: &'a LocalWaker, spawner: &'a mut dyn Spawn) -> Context<'a> { ... }
+
+    pub fn spawner(&mut self) -> &mut dyn Spawn { ... }
+
+    pub fn with_spawner<'b, Sp: Spawn>(&'b mut self, spawner: &'b mut Sp) -> Context<'b> { ... }
+
+    // ...
+}
+```
+
+### Spawn macros
+
+This release of futures-rs introduces the `spawn!` and `spawn_with_handle!` macros for use in async functions, closures and blocks. Both macros let you spawn a task that polls the given future to completion. The task is spawned via the current context's spawner.
+
+`spawn!` lets you spawn futures with `Output = ()`:
+
+```rust
+#![feature(async_await, await_macro, futures_api)]
+use futures::spawn;
+
+spawn!(future).unwrap();
+```
+
+`spawn_with_handle!` lets you spawn futures with any `Output: Send` type and also lets you await the produced output value:
+
+```rust
+use futures::spawn_with_handle;
+
+let join_handle = spawn_with_handle!(future).unwrap();
+let output = await!(join_handle);
+```
+
+`spawn!` is slightly cheaper than `spawn_with_handle!` since it doesn't need to provide a way to get the output.
+
+Should spawning fail, a `SpawnError` error is returned which contains information about why it failed.
+
+So what futures can be spawned? There are some requirements: The future needs to be `Send` and `'static` and the output needs to be `Send`. Here's why:
+- The `'static` lifetime means that the future cannot contain any non-static (exterior) references. This is required so that the spawned task can be run independently to the task that spawns it.
```

Comment:
I'd remove "exterior" here-- I don't think it adds anything. Rust users know what `'static` is.

Reply:
Done.

What I wanted to say is that it could contain interior references and still be `'static`. The future doesn't at that stage, though, because it isn't pinned yet. So you're right the distinction isn't helpful.